File: src/jvmMain/java/RideOrder.java
import java.time.LocalDateTime;

public class RideOrder {
    private int id;
    private String passengerName;
    private String destination;
    private double price;
    private LocalDateTime orderTime; // Обов'язкове поле дати та часу

    public RideOrder(int id, String passengerName, String destination, double price) {
        this.id = id;
        this.passengerName = passengerName;
        this.destination = destination;
        this.price = price;
        this.orderTime = LocalDateTime.now(); // Автоматично фіксуємо час створення
    }

    // Гетери та сетери
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getPassengerName() { return passengerName; }
    public void setPassengerName(String passengerName) { this.passengerName = passengerName; }

    public String getDestination() { return destination; }
    public void setDestination(String destination) { this.destination = destination; }

    public double getPrice() { return price; }
    public void setPrice(double price) { this.price = price; }

    public LocalDateTime getOrderTime() { return orderTime; }

    @Override
    public String toString() {
        return "Замовлення #" + id +
                " | Пасажир: " + passengerName +
                " | Пункт призначення: " + destination +
                " | Ціна: ₴" + price +
                " | Час: " + orderTime;
    }
}

File: src/commonMain/kotlin/RideOrder.kt
import kotlinx.datetime.Clock
import kotlinx.datetime.LocalDateTime
import kotlinx.datetime.TimeZone
import kotlinx.datetime.toLocalDateTime

// Data class у Kotlin автоматично створює гетери, сетери та toString()
data class RideOrder(
    val id: Int,
    val passengerName: String,
    val destination: String,
    val price: Double
) {
    // Фіксуємо час створення через кросплатформний kotlinx-datetime
    val orderTime: LocalDateTime = Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault())
}